app/main.rb

```diff
@@ -0,0 +1,153 @@
+require_relative '../classes/student'
+require_relative '../classes/teacher'
+require_relative '../classes/book'
+require_relative '../classes/rental'
+
+def choices
+  {
+    1 => 'List all books',
+    2 => 'List all people',
+    3 => 'Create a person',
+    4 => 'Create a book',
+    5 => 'Create a rental',
+    6 => 'List all rentals for a person id',
+    7 => 'Exit'
+  }
+end
+
+# State class to store app's state
+class State
+  def initialize
+    @books = []
+    @people = []
+  end
+
+  def create_student
+    print 'Age:'
+    age = gets.chomp
+    print 'Name:'
+    name = gets.chomp
+    print 'Has parent permission?[Y/N]:'
+    permission = gets.chomp
+    @people << Student.new(age, name, parent_permission: permission.upcase == 'Y')
+    puts 'Student added successfully'
+  end
+
+  def create_teacher
+    print 'Age:'
+    age = gets.chomp
+    print 'Name:'
+    name = gets.chomp
+    print 'Specialization:'
+    specialization = gets.chomp
+    @people << Teacher.new(specialization, name, age)
+  end
+
+  def create_book
+    print 'Title:'
+    title = gets.chomp
+    print 'Author:'
+    author = gets.chomp
+    @books << Book.new(title, author)
+    puts 'Book crteated successfully'
+  end
+
+  def list_people(indexed: false)
+    @people.each_with_index do |person, index|
+      print "#{index}) " if indexed
+      print "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age:#{person.age}\n"
+    end
+    main_menu
+  end
+
+  def list_books(indexed: false)
+    @books.each_with_index do |book, index|
+      print "#{index}) " if indexed
+      print "Title: #{book.title}, Author: #{book.author}\n"
+    end
+    main_menu
+  end
+
+  def create_rental
+    if @people.size.zero? || @books.size.zero?
+      return 'You didn\'t have any person and/or book added yet; rental cant be created!'
+    end
+
+    puts 'Select a book from the following list by number'
+    list_books(indexed: true)
+    book = gets.chomp
+
+    puts 'Select a person from the following list by number'
+    list_people(indexed: true)
+    person = gets.chomp
+
+    print 'Date:'
+    date = gets.chomp
+    Rental.new(date, @books[book.to_i], @people[person.to_i])
+
+    'Rental created successfully'
+  end
```

Comment:
- [ ] Kindly make sure your option(5) for creating rentals is working, Kindly make sure you fix the scenario when the user selects a person with the 0, your app should not exit rather it should create the appropriated rental.

![image](https://user-images.githubusercontent.com/65068771/193159543-b98b4b69-2dc3-4e7d-9a8d-ae38994bdff5.png)

Reply:
Done